tools/jira_notificator.py

```diff
@@ -0,0 +1,164 @@
+import logging
+
+from enum import Enum
+from typing import Optional
+from jira import JIRA, Issue
+from jira.resources import User
+from datetime import datetime, timedelta, timezone
+from dateutil import parser
+from oar.core.ldap import LdapHelper
+
+logger = logging.getLogger(__name__)
+
+ERT_NOTIFICATION_PREFIX = "Errata Reliability Team Notification"
+
+class NotificationType(Enum):
+    QA_CONTACT = (24, "QA Contact Action Request")
+    TEAM_LEAD = (48, "Team Lead Action Request")
+    MANAGER = (72, "Manager Action Request")
+
+    def __init__(self, hours: int, label: str):
+        self.hours = hours
+        self.label = label
+
+def create_notification_prefix(notification_type: NotificationType) -> str:
+    return f"{ERT_NOTIFICATION_PREFIX} - {notification_type.label}: This issue has been in the ON_QA state for over {notification_type.hours} hours."
+
+def get_notification_type(notification_message: str) -> Optional[NotificationType]:
+    for notification_type in NotificationType:
+        if notification_type.label in notification_message:
+            return notification_type
+    return None
+
+def find_user_by_email(jira: JIRA, email) -> Optional[User]:
+    for user in jira.search_users(user=email):
+        if user.emailAddress == email:
+            return user
+    logger.warning(f"User was not found for email {email}.")
+    return None
+
+def get_qa_contact(issue: Issue) -> Optional[User]:
+    qa_contact = issue.fields.customfield_12315948
+    if qa_contact:
+        return qa_contact
+    else:
+        logger.warning(f"Issue {issue.key} does not have QA contact.")
+        return None
+
+def get_manager(jira: JIRA, qa_contact: User) -> Optional[User]:
+    ldap = LdapHelper()
+    manager_email = ldap.get_manager_email(qa_contact.emailAddress)
+    if manager_email:
+        manager = find_user_by_email(jira, manager_email)
+        if manager:
+            return manager
+        else:
+            logger.warning("Notification to QA Contact Manager could not be sent because manager user was not found in Jira.")
+    
+    else:
+        logger.warning("Notification to QA Contact Manager could not be sent because QA contact manager was not found.")
+    return None
+
+def create_qa_notification_text(issue: Issue)  -> str:
+    qa_contact = get_qa_contact(issue)
+    qa_contact_link = f"[~{qa_contact.name}] " if qa_contact else ""
+    return f"{create_notification_prefix(NotificationType.QA_CONTACT)} {qa_contact_link}Please verify the Jira as soon as possible."
+
+def notify_qa_contact(jira: JIRA, issue: Issue) -> None:
+    jira.add_comment(issue, create_qa_notification_text(issue))
```

Comment:
in func `notify_xxx` need to handle that cases like qa_contact or manager id not found

Reply:
Current logic is: if qa contact or manager is not found the link to jira user is not added to the message. So the notification is just a text. 

I not sure what we can do with theses jiras, maybe we can notify members of our team in Jira/Slack to take a look and manually assign. WDYT?

Reply:
The new logic is now:
If QA contact or manager is not found, assignee and manager of assignee are asked for help.
If manager of assignee is not found in Jira, only assignee is notified
If Issues does not have assignee, raise error (and continue with next issue).